Simplenote/Classes/SPCardViewController.swift

```diff
@@ -0,0 +1,170 @@
+import UIKit
+
+// MARK: - SPCardViewController wraps passed viewController in a view with rounded corners and a shadow
+//
+final class SPCardViewController: UIViewController {
+    private let viewController: UIViewController
+    private let containerView = UIView()
+
+    private var bottomToBottomConstraint: NSLayoutConstraint?
+    private var topToBottomConstraint: NSLayoutConstraint?
+
+    init(viewController: UIViewController) {
+        self.viewController = viewController
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        stopListeningToNotifications()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = .clear
+
+        addShadowView()
+        setupContainerView()
+        addChildViewController()
+
+        refreshStyle()
+
+        startListeningToNotifications()
+    }
+}
+
+// MARK: - Presentation
+//
+extension SPCardViewController {
+
+    /// Presents card from another view controller animated
+    ///
+    func present(from viewController: UIViewController) {
+        viewController.addChild(self)
+
+        view.translatesAutoresizingMaskIntoConstraints = false
+        let containerView = viewController.view!
+        containerView.addSubview(view)
+
+        setupConstraints(withContainerView: containerView)
+
+        containerView.layoutIfNeeded()
+
+        let animations = { () -> Void in
```

Comment:
Indeed, thanks 👍 
Funny thing is that compiler doesn't always correctly infer return type in case it's a one-liner.
For example in the following code compiler thinks that the signature is `() -> ()?` because of the optional `snapshot` variable...
```
let animations = {
    snapshot?.alpha = 0.0
}
```

Reply:
Ahhhhh TIL! 

Right! Single statement closures are evaluated, and the compiler infers you're actually returning the value.